tests/testthat/test-classSelection.R

```diff
@@ -0,0 +1,133 @@
+  folds <- lapply(folds, function(x) setdiff(seq_along(Y.train), x))
+  fold.idx <- sort(unique(unlist(folds)))
+  expect_true(all(fold.idx == seq_along(Y.train)), "CV indexes not generated correctly")
+
+  # Train a caret ensemble
+  ctrl <- trainControl(method="cv", savePredictions="final", classProbs=TRUE, index=folds)
+  model.list <- caretList(
+    X.train, Y.train, metric = "Accuracy",
+    trControl = ctrl, methodList = c("rpart", "glmnet"))
+  model.ens <- caretEnsemble(model.list)
+
+  # Verify that the observed responses in each fold, for each model,
+  # have the same levels and that the first level is equal to the first
+  # level in the original data (i.e. Y.class).  This check exists to
+  # avoid regressions to bugs like this:
+  # https://github.com/zachmayer/caretEnsemble/pull/190
+  unique.levels <- unique(sapply(model.ens$models, function(x) levels(x$pred$obs)[1]))
+  expect_identical(unique.levels, Y.levels[1])
+
+  # Verify that the training data given to the ensemble model has the
+  # same levels in the response as the original, raw data
+  expect_identical(levels(model.ens$ens_model$trainingData$.outcome), Y.levels)
+
+  # Create class and probability predictions, as well as class predictions
+  # generated from probability predictions using a .5 cutoff
+  Y.pred <- predict(model.ens, newdata=X.test, type="raw")
+  Y.prob <- predict(model.ens, newdata=X.test, type="prob")
+  Y.cutoff <- factor(ifelse(Y.prob > .5, Y.levels[pos.level], Y.levels[-pos.level]), levels=Y.levels)
+
+  # Create confusion matricies for each class prediction vector
+  cmat.pred <- confusionMatrix(Y.pred, Y.test, positive=Y.levels[pos.level])
+  cmat.cutoff <- confusionMatrix(Y.cutoff, Y.test, positive=Y.levels[pos.level])
+
+  # Verify that the positive level of the Y response is equal to the positive
+  # class label used by caret.  This could potentially become untrue if
+  # the levels of the response were ever rearranged by caretEnsemble at some point.
+  expect_identical(cmat.pred$positive, Y.levels[pos.level])
+
+  # Verify that the accuracy score on predicted classes is relatively high.  This
+  # check exists to avoid previous errors where classifer ensemble predictions were
+  # being made using the incorrect level of the response, causing the opposite
+  # class labels to be predicted with new data.
+  expect_equal(as.numeric(cmat.pred$overall["Accuracy"]), 0.7586, tol = 0.0001)
+
+  # Similar to the above, ensure that probability predictions are working correctly
+  # by checking to see that accuracy is also high for class predictions created
+  # from probabilities
+  expect_equal(as.numeric(cmat.cutoff$overall["Accuracy"]), 0.7586, tol = 0.0001)
+}
+
+test_that("Ensembled classifiers do not rearrange outcome factor levels", {
+  skip_on_cran()
+
+  # Make sure that caretEnsemble uses the first level in the
+  # outcome factor as the target class
+  options(caret.ensemble.target.bin.level=1)
```

Comment:
You should get the original option, and then re-set it at the end of the test, e.g.

```{R}
oldOption <- getOption('caret.ensemble.target.bin.level=')
...
options(caret.ensemble.target.bin.level=oldOption)
```